### PR TITLE
Trace all HTTP codes other than 200 as errors

### PR DESCRIPTION
**Pull request by vlad-ivanov-name**
This allows to raise logging level but still see 500, 404 errors etc

### CODE REVIEW

**Comment by LMG** [comment]
Looks good but you forgot to check format!

File: josh-proxy/src/bin/josh-proxy.rs [hunk 1]
```diff
@@ -678,10 +678,15 @@ async fn run_proxy() -> josh::JoshResult<i32> {
                     error_response().await
                 };
                 let _e = s.enter();
+                let http_status = r.status().as_u16()
+                let log_level = match http_status {
+                    200 => tracing::Level::TRACE,
```
Comment (christian-schilling) [inline josh-proxy/src/bin/josh-proxy.rs:683]: I would prefer something like:
```
< 400 -> TRACE // All different kinds of success
< 500 -> WARN // Something wrong with the request, clients fault
_ -> ERROR // Server side error, need to debug josh
```

Comment (christian-schilling) [reply josh-proxy/src/bin/josh-proxy.rs:683]: Actually, the first should probably be <= 401, as "unauthorized" is a very common and normal response that will happen pretty much all the time without any need for concern.

Comment (vlad-ivanov-name) [reply josh-proxy/src/bin/josh-proxy.rs:683]: @christian-schilling but it also returns 500 here when checkout URL is malformed

Comment (vlad-ivanov-name) [reply josh-proxy/src/bin/josh-proxy.rs:683]: ah I think it still makes sense. Thanks, I'll update the PR a bit later

Comment (christian-schilling) [reply josh-proxy/src/bin/josh-proxy.rs:683]: If it returns 500 when the URL is malformed that's a bug in itself, it should be something 4xx.

Comment (vlad-ivanov-name) [reply josh-proxy/src/bin/josh-proxy.rs:683]: image

**Review by christian-schilling — APPROVED** [review]

**Pull request merged**